sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFilterSuite.scala

```diff
@@ -330,9 +331,56 @@ class ParquetFilterSuite extends QueryTest with ParquetTest with SharedSQLContex
 
         // If the "c = 1" filter gets pushed down, this query will throw an exception which
         // Parquet emits. This is a Parquet issue (PARQUET-389).
+        val df = sqlContext.read.parquet(pathOne, pathTwo).filter("c = 1").selectExpr("c", "b", "a")
         checkAnswer(
-          sqlContext.read.parquet(pathOne, pathTwo).filter("c = 1").selectExpr("c", "b", "a"),
+          df,
           (1 to 1).map(i => Row(i, i.toString, null)))
+
+        // The fields "a" and "c" only exist in one Parquet file.
+        df.schema.fields.foreach { f =>
+          if (f.name == "a" || f.name == "c") {
+            assert(f.metadata.contains("optional"))
+          }
+        }
```

Comment:
I'd prefer the following version. If `a` or `c` doesn't even exist in `df.schema`, the current version still passes.

```scala
assert(df.schema("a").metadata.contains("optional"))
assert(df.schema("c").metadata.contains("optional"))
```